design/object-store.md

```diff
@@ -0,0 +1,141 @@
+
+### Pools
+
+The pools are the backing data store for the object store and are created with specific names to be private to an object store. Pools can be configured with all of the settings that can be specified in the [Pool CRD](/Documentation/pool-crd.md). The underlying schema for pools defined by a pool CRD is the same as the schema under the `metadataPool` and `dataPool` elements of the object store CRD. All metadata pools are created with the same settings, while the data pool can be created with independent settings. The metadata pools must use replication, while the data pool can use replication or erasure coding. 
+
+```yaml
+  metadataPool:
+    failureDomain: host
+    replicated:
+      size: 3
+  dataPool:
+    failureDomain: device
+    erasureCoded:
+      dataChunks: 6
+      codingChunks: 2
+```
+
+### Gateway
+
+The gateway settings correspond to the RGW service. 
+- The `type` can be `S3` or `Swift`
+- If an ssl certificate is available, the RGW service will start with SSL termination enabled
+- The `port` is where the RGW pods and the RGW service will be listening
+- The `replicas` are the number of pods that will be started for this object store
+- The rgw pods can be given standard Kubernetes placement restrictions with the `placement` setting.
+
+```yaml
+ gateway:
+    type: s3
+    sslCertificateRef: my-ssl-cert-secret
+    port: 53390
+    replicas: 1
+    placement:
+```
+
+### Realms, Zone Groups, and Zones
+
+By default, the object store will be created independently from any other object stores and replication to another object store will not be configured. This done by creating a new realm, zone group, and zone all with the name of the new object store. The zone group and zone are tagged as the `master`. If this is the first object store in the cluster, the realm, zone group, and zone will also be marked as the default.
+
+By implementing on the independent realms, zone groups, and zones, Rook supports multiple objects stores in a cluster. The set of users with access to the object store, the metadata, and the data are isolated from other object stores. 
+
+If desired to configure the object store to replicate from another cluster or zone, the following settings would be specified on a new object store that is *not* the master. (This feature is not yet implemented.)
+- If `realm` is specified, the new zone will be created in the existing realm with that name
+- If `group` is specified, the new zone will be created in the existing zone group with that name
+- The `master` settings indicate the RGW endpoint where this object store will need to connect to the other cluster or zone in order to initialize the replication. The operator will execute `pull` commands for the realm and zone group as necessary.
+```yaml
+  zone:
+    realm: myrealm
+    group: mygroup
+    master:
+      url: https://my-master-zone-gateway:443/
```

Comment:
how does DNS work here? does the k8s service that we create have a DNS entry for it too?

Reply:
If it is in the same cluster then yes the K8s service can be used. If the master is in another K8s cluster, the network admin will have to manage the dns entries.